### PR TITLE
Issue 242: add an @interfaces expandable endpoint

### PR DESCRIPTION
**Pull request by erral**
See #242

### CODE REVIEW

**Comment by buchi** [comment]
👍 LGTM

I wonder if exposing all those interface names publicly may lead to unwanted information disclosure.

**Comment by erral** [comment]
Yeah... during the training we wondered if we could restrict somehow the list of exposed interfaces, or at least set a list of known-good-set of interfaces, and expose just them...

File: docs/source/interfaces.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Interfaces
+==========
+
+Getting the interfaces provided by the current context:
```
Comment (tisto) [inline docs/source/interfaces.rst]: @erral we usually try to add a short introduction to non-Plone devs what this documentation section is about. E.g. a short introduction what interfaces are, maybe with a link to the Plone docs.

**Comment by tisto** [comment]
@erral LGTM as well. Though, as @buchi said, the interfaces are currently not visible by anonymous users. Therefore we might have to think about this...

**Comment by erral** [comment]
I share your concerns about the public visibility of the interfaces. Nevertheless they are needed, at least to know the navigation contexts when the front-end wants to build the global navigation for instance.

@ebrehault any comment on this?

**Comment by tkimnguyen** [comment]
What would you recommend as a way of restricting who has access to this?

**Comment by jensens** [comment]
So, what is the problem with exposing? 

Does exposing information about interfaces - and so about installed add-ons and other internals - give attackers hints about possible vulnerabilities outside of the core? But also FTI info and other endpoint expose that information. 

I doubt it a problem, but I would love to hear opposing opinions.

**Comment by mauritsvanrees** [comment]
In PloneHotfix20161129 we removed the docstrings of lots of zope.interface methods and attributes to avoid 'publishing' them, that is making them available for browsers to call. See https://github.com/zopefoundation/Zope/pull/81. In some cases they could really be used for potential harm (setting values), in others it was: better safe than sorry.

Just giving the name of interfaces *should* be fine. But it *does* give you info about an object that is useful for a UI but also for potential hackers.
So a way to whitelist some interfaces, or to completely disable this (return an empty list) for the security conscious would be good.

**Comment by jensens** [comment]
What is the state here? Do we want it or not? If not and hence there was no activity for a long time, I propose to close this PR within next two weeks. If you do not feel OK with this, please speak up and provide us a rough schedule.

**Comment by tisto** [comment]
@jensens it is something we definitely want. Though, there are lots of things to consider and to discuss. I'd prefer to keep it open. Maybe something that can be discussed at one of the next sprints this year...

**Comment by jensens** [comment]
This one got forgotten. Is it still valid or meanwhile superseded?

**Comment by erral** [comment]
I am closing this because the main reason for this, the navigation root thing, has been fixed using another endpoint. If we need to expose the interfaces in the future we can use this code as an inspiration, but we do not need this right now.

**Pull request closed without merging**